core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -385,11 +385,9 @@ SharedAllocationRecord<void, void>
   template class Kokkos::Impl::HostInaccessibleSharedAllocationRecordCommon<           \
       MEMORY_SPACE>
 
-namespace {
-
 /* Taking the address of this function so make sure it is unique */
 template <class MemorySpace, class DestroyFunctor>
-void deallocate(SharedAllocationRecord<void, void>* record_ptr) {
+inline void deallocate(SharedAllocationRecord<void, void>* record_ptr) {
```

Comment:
Any idea why we had the anonymous namespace?
The `inline` specifier is not necessary here.  (Not blocking but I would like to understand the change)

Reply:
I left the `inline` here to signal intent that we only want one symbol generated (since we are later taking the address of this function). I don't know why an `anonymous` namespace was used instead (generating one symbol per compilation unit). The `function` sanitizer complained here.

Reply:
but you disabled the "function" diagnostic in the end so why do we want this resolved here?

Reply:
This was one fix I could make (even if we won't diagnose it again with the current set of checks).